server/api/dev.js

```diff
@@ -0,0 +1,214 @@
+        users.push({ username: 'dev' });
+      }
+
+      /*
+        ------- ENTRIES -------
+        Create some flags for yourself and dev user
+        Adust as desired
+        NOTE: borrows flags from existing entires but overrides key data
+              accurate url, width, height
+              diversity of images, descriptions, etc
+      */
+      let entryVersions = [
+        { user: 'dev', name: 'Pending example', submission_status: 'pending' },
+        { user: 'dev', name: 'Withdrawn example', submission_status: 'withdrawn' },
+        {
+          user: 'dev', name: 'Approved example', submission_status: 'approved', modified_by: 'VertigoOne',
+        },
+        {
+          user: 'dev', name: 'Rejected example', submission_status: 'rejected', modified_by: 'VertigoOne', rejection_reason: 'Rejected because.',
+        },
+        { user: username, name: 'Pending example', submission_status: 'pending' },
+        { user: username, name: 'Withdrawn example', submission_status: 'withdrawn' },
+        {
+          user: username, name: 'Approved example', submission_status: 'approved', modified_by: 'VertigoOne',
+        },
+        {
+          user: username, name: 'Rejected example', submission_status: 'rejected', modified_by: 'VertigoOne', rejection_reason: 'Rejected because.',
+        },
+      ];
+
+      // add sequential dev# for id (NOTE: zero-indexed, can do i+1 if desired)
+      entryVersions = entryVersions.map((v, i) => ({ ...v, id: `dev${i}`, background_color: BACKGROUND_COLORS[i % 3] }));
+
+      const existingEntries = await db.select(`SELECT width, height, url FROM entries WHERE height > 600 AND POSITION('https://firebasestorage.googleapis.com' IN url) > 0 LIMIT ${entryVersions.length}`);
+      await db.insert('entries', entryVersions.map((version, i) => ({
+        ...existingEntries[i], ...version,
+      })));
+
+      /*
+        ------- CONTEST_ENTRIES -------
+        Connect the two
+      */
+      await db.insert('contest_entries', entryVersions.map(({ id }, i) => ({
+        contest_id: 'dev',
+        entry_id: id,
+        rank: null,
+        category: categories[i % categories.length].category, // alternating entries
+      })));
+
+      /*
+        ------- VOTES -------
+        Random voting, all users on all flags
+      */
+      const approvedEntries = entryVersions.filter((version) => version.submission_status === 'approved');
+
+      await db.insert('votes', users.flatMap(({ username: u }) => (
+        approvedEntries.map(({ id }) => ({
+          username: u,
+          entry_id: id,
+          contest_id: 'dev',
+          rating: u === approvedEntries.user
+            ? VOTE_MAX
+            : Math.floor(Math.random() * (VOTE_MAX + 1)),
+          last_modified: NOW,
+        }))
+      )));
+    } else {
+      const contest = getDevContest(status);
+      // Switch contest status
+      if (isExisting) {
+        // update rejects date for some reason
```

Comment:
I think you need to cast the date in to a string for it to work.

Reply:
Idk, `toISOString()` still doesn't pass here.

Reply:
From what I can tell ([StackOverflow](https://stackoverflow.com/questions/75485625/is-it-possible-to-enum-and-timestamp-using-pgp-helpers-update)), you need to define the cast in your column set. So it could be something like:
```javascript
await db.update('contests', [contest], Object.entries(contest).map(([key, value]) => {
  if (key === 'id') {
    return '?id';
  }
  if (value instanceof Date) {
    return { name: key, cast: 'timestamp' };
  }
  return key;
});
```